cores/esp32/esp32-hal-spi.c

```diff
@@ -177,36 +177,73 @@ static spi_t _spi_bus_array[] = {
 static bool spiDetachBus(void * bus){
     uint8_t spi_num = (int)bus - 1;
     spi_t * spi = &_spi_bus_array[spi_num];
-    if(spi->dev->clock.val != 0){
-        log_d("Stopping SPI BUS");
+
+    if(spi->dev->clock.val == 0){
+        log_d("SPI bus already stopped");
+        return true;
+    }
+    else if(spi->sck == -1 || (spi->miso == -1 && spi->mosi == -1)){
+        log_d("Stopping SPI bus");
         spiStopBus(spi);
+
+        if(spi->sck != -1){
+            spiDetachSCK(spi);
+        }
+        if(spi->miso != -1){
+            spiDetachMISO(spi);
+        }
+        if(spi->mosi != -1){
+            spiDetachMOSI(spi);
+        }
+        if(spi->ss != -1){
+            spiDetachSS(spi);
+        }
```

Comment:
The checks for -1 can be removed as they are already done inside the function.
```suggestion
        spiDetachSCK(spi);
        spiDetachMISO(spi);
        spiDetachMOSI(spi);
        spiDetachSS(spi);
```